### ISSUE TITLE
stylelint fail in typescript file containing generic type for function

### ISSUE THREAD

**Issue by huy-nguyen** [styled-components/stylelint-processor-styled-components#83]
This is a minimal example of a TypeScript file that caused `stylelint` to fail:

```typescript
const someFunction = <T>(input: T) => input;
```
Linting succeeds after the generic type `T` is removed:
```typescript
const someFunction = (input: number) => input;
```

`stylelint` config used to lint that file:
```js
module.exports = {
  processors: ['stylelint-processor-styled-components'],
  extends: [
    'stylelint-config-standard',
  ],
  syntax: 'scss',
  rules: {
    'declaration-block-semicolon-newline-after': null,
    'rule-empty-line-before': null,
    'declaration-block-trailing-semicolon': null,
  },
}
```
Versions of relevant packages:
- `typescript`: 2.4.0
- `stylelint-processor-styled-components`: 0.2.2.
- `typescript-eslint-parser`: 5.0.1
- `stylelint`: 8.0.0

**Comment by emilgoldsmith** [comment]
Hey, did this also happen with the previous version of `typescript-eslint-parser` ? As in, updating that package was the only thing we did in v0.2.2 so a more direct question may be, does this also occur with v0.2.1?

**Comment by huy-nguyen** [comment]
Yes the same problem happens with `stylelint-processor-styled-components` 0.2.1.

**Comment by emilgoldsmith** [comment]
@huy-nguyen Okay, good to know! If you want to help us speed up the process submitting a PR with a failing test is always helpful (and of course a fix would be even more helpful!)

@styled-components/stylelint-processor I've never used Typescript before and have a few other things that are on my to-do list in this repo when I make more time to work on it, so I won't be making this issue a priority personally, just FYI.

**Comment by emilgoldsmith** [comment]
@ismay are you familiar with Typescript? I'd love to have this issue out of the way by 1.0 as it seems to be a bug. I never used Typescript so I'd love to have someone else take a look at this for me.

**Comment by ismay** [comment]
@emilgoldsmith Unfortunately not. Have a little experience with flow, but not with typescript.

**Comment by mxstbr** [comment]
It seems unlikely that this is something on our side and sounds like a bug in the parser, so I'm going to close this issue.

It'd be great if you could [submit an issue over at `typescript-eslint-parser`](https://github.com/eslint/typescript-eslint-parser/issues/new), linking them to [the file where we use it](https://github.com/styled-components/stylelint-processor-styled-components/blob/976f8d5221f35f31dc35ced324b855eb70ec9657/src/parsers/typescript-parser.js) and posting your full error/stack trace! I'm sure they'd love to help you, /cc @azz

**Comment by fabb** [comment]
Maybe related: #274
